pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,451 @@
+	NoFlag = 100 // Default value if no flag is specified (dummy, we use cmd.Flags().Changed() to check if a flag is specified)
+	noArg  = 10  // Default value if no argument specified (e.g. "--virt-api" = "--virt-api=10")
+	// verbosity must be 0-9
+	// https://kubernetes.io/docs/reference/kubectl/cheatsheet/#kubectl-output-verbosity-and-debugging
+	minVerbosity = uint(0)
+	maxVerbosity = uint(9)
+)
+
+// Log verbosity can be set per KubeVirt component
+// https://kubevirt.io/user-guide/operations/debug/#setting-verbosity-per-kubevirt-component
+// TODO: set verbosity per nodes
+type virtComponent int
+
+// also used by the test file
+const (
+	VirtAPI virtComponent = iota // virtAPI must be at the first position because it is used for the iteration
+	VirtController
+	VirtHandler
+	VirtLauncher
+	VirtOperator
+	All // all must be at the end, because it is used for the iteration
+)
+
+// also used by the test file
+const VirtComponentNum = int(All) + 1 // number of virt components
+
+// for receiving the flag argument
+var verbosities [VirtComponentNum]uint
+var isReset bool
+
+// operation type of log-verbosity command
+type operation int
+
+const (
+	show operation = iota
+	set
+	nop
+)
+
+// for patch operation
+const (
+	// just "add" is fine, no need of "replace" and "remove"
+	// https://www.rfc-editor.org/rfc/rfc6902
+	patchOperation = patch.PatchAddOp
+	patchPath      = "/spec/configuration/developerConfiguration/logVerbosity"
+)
+
+func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "log-verbosity",
+		Short: "Show or Set/Reset log verbosity. The verbosity must be 0-9. Show and Set/Reset cannot coexist.\n",
+		Long: "- To show the log verbosity of one or more components " +
+			"(when the log verbosity is unattended in the KubeVirt CR, show the default verbosity (2)).\n" +
+			"- To set the log verbosity of one or more components.\n" +
+			"- To reset the log verbosity of all components " +
+			"(empty the log verbosity field, which means reset to the default verbosity (2)).\n\n" +
+			"- The components are <virt-api|virt-controller|virt-handler|virt-launcher|virt-operator>.\n" +
+			"- Show and Set/Reset cannot coexist.\n" +
+			"- Verbosity must be 0-9.\n" +
+			"- Flag syntax must be \"flag=arg\" (\"flag arg\" not supported).",
+		Example: usage(),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			c := Command{command: "log-verbosity", clientConfig: clientConfig}
+			return c.RunE(cmd)
+		},
+	}
+
+	cmd.Flags().UintVar(&verbosities[VirtAPI], "virt-api", NoFlag, "show/set virt-api log verbosity (0-9)")
+	// Set default value (noArg) if the flag has no argument, because we use the flag without an argument (e.g. --virt-api) to show verbosity
+	// Otherwise, the pflag package will return an error due to missing argument
+	cmd.Flags().Lookup("virt-api").NoOptDefVal = strconv.FormatUint(noArg, 10)
```

Comment:
Why 10? Would be possible to create a constant and replace the 10 here and below?

Reply:
The second argument of `strconv.FormatUint()` is the base number (10 means base 10).

`cmd.Flags().Lookup().NoOptDefVal` is a string. So we need to convert `uint` to `string`, and here we use `strconv.FormatUint()`.

We can create a constant, but this is the base number, so I think it is ok to use the literal. But I do not have a strong preference, so I am also ok to use a constant for this number (like `const base = 10`). Please let me know if you prefer to use a constant.

OR, I think the following might be a better approach.

I used `strconv.FormatUint()` because
- `Strconv.FormatUint()` was the only way I could find to convert uint to string. 
- Someone has measured that `strconv.FormatInt()` is slightly faster than `strconv.Itoa()`.

However, in our case, `Itoa()` would be fine because we know that `noFlag` is always 10, and the execution speed is not a big deal. `Itoa()` may be simpler. So if we do not mind that noArg is actually uint, I will change this to use `strconv.Itoa()`. If you do not want to use `strconv.Itoa()`, please let me know.

Reply:
Ok sorry, you are right

Reply:
Thanks for the confirmation. Then, I’ll leave the code as is (i.e. keep using strconv.FormatUint()).